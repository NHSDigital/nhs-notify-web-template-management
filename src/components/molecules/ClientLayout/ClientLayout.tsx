'use client';

import '@styles/app.scss';
import { NHSNotifyHeader } from '@molecules/Header/Header';
import { NHSNotifyContainer } from '@layouts/container/container';
import { NHSNotifyFooter } from '@molecules/Footer/Footer';
import { NHSNotifySkipLink } from '@atoms/NHSNotifySkipLink/NHSNotifySkipLink';
import { NHSNotifyAuthenticator } from '@molecules/NHSNotifyAuthenticator/NHSNotifyAuthenticator';

export const ClientLayout = ({ children }: { children: React.ReactNode }) => {
  return (
    <NHSNotifyAuthenticator>
      <NHSNotifySkipLink />
      <NHSNotifyHeader />
      <NHSNotifyContainer>{children}</NHSNotifyContainer>
      <NHSNotifyFooter />
    </NHSNotifyAuthenticator>
  );
};
