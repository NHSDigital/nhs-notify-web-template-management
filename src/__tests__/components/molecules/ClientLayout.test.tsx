import { render, waitFor } from '@testing-library/react';
import { ClientLayout } from '@molecules/ClientLayout/ClientLayout';

test('ClientLayout', async () => {
  const container = render(
    <ClientLayout>
      <p>testing</p>
    </ClientLayout>
  );

  await waitFor(() => {});
  expect(container.asFragment()).toMatchSnapshot();
});
