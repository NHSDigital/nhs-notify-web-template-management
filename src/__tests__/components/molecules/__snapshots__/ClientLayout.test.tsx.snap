// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClientLayout 1`] = `<DocumentFragment />`;
